### PR TITLE
Allow to define the signer from the configuration.

### LINKED ISSUE #28: Add the ability to chose the signer to use from the configuration.
This will be useful in order to use either the ECDSA local signer or the remote Autograph signer.

### PR DESCRIPTION
**Pull request by almet**
Fixes #28.

r? @leplatrem @Natim

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -60,20 +70,44 @@ to be set.
...
+Configuration for the (default) ECDSA local signer
+--------------------------------------------------
+
++---------------------------------+--------------------------------------------------------------------------+
+| Setting name                    | What does it do?                                                         |
++=================================+==========================================================================+
+| kinto_signer.ecdsa.private_key  | The absolute path to the location of the ECDSA private key to use to     |
 |                                 | apply the signatures                                                     |
 +---------------------------------+--------------------------------------------------------------------------+
+| kinto_signer.ecdsa.public_key   | The absolute path to the location of the ECDSA private key to use to     |
```
Comment (leplatrem) [inline README.rst]: `absolute path to the location` sounds weird. Why not `absolute path to the ... private key file` ?

File: README.rst [hunk 2]
```diff
@@ -60,20 +70,44 @@ to be set.
...
+use the following settings:
+
++------------------------------------+--------------------------------------------------------------------------+
+| Setting name                       | What does it do?                                                         |
++====================================+==========================================================================+
+| kinto_signer.autograph.hawk_id     | The hawk identifier used to issue the requests.                          |
++------------------------------------+--------------------------------------------------------------------------+
+| kinto_signer.autograph.hawk_secret | The hawk secret used to issue the requests.                              |
++------------------------------------+--------------------------------------------------------------------------+
+| kinto_signer.autograph.server_url  | The URL on which the autograph server is listening for incoming requests |
```
Comment (leplatrem) [inline README.rst]: I would put this setting on top and rephrase to `The autograph server URL`, which is simple and obvious

File: kinto_signer/hook.py [hunk 3]
```diff
@@ -2,45 +2,54 @@
...
 
 
 def includeme(config):
-    # Process settings to remove storage wording.
     settings = config.get_settings()
 
-    config.registry.signer = AutographSigner(settings)
+    # Load the signer from its dotted location. Fallback to the local ECDSA
+    # signer.
+    ecdsa_module_location = "kinto_signer.signer.local_ecdsa"
```
Comment (leplatrem) [inline kinto_signer/hook.py]: nit: `default_signer_location` ?

File: kinto_signer/signer/local_ecdsa.py [hunk 4]
```diff
@@ -10,35 +10,36 @@
 
 class ECDSASigner(object):
 
-    def __init__(self, settings=None):
-        self.settings = settings or {}
+    def __init__(self, private_key=None, public_key=None):
+        if private_key is None and public_key is None:
+            msg = ("Please, specify either a private_key or public_key "
+                   "location.")
+            raise ValueError(msg)
```
Comment (leplatrem) [inline kinto_signer/signer/local_ecdsa.py]: There is `ConfigurationError` in `pyramid.exceptions`, maybe you want to use it.

Comment (almet) [inline kinto_signer/signer/local_ecdsa.py]: We already had this discussion I believe: I want to keep the signers separate form pyramid as much as possible.

Comment (leplatrem) [inline kinto_signer/signer/local_ecdsa.py]: haha yes, now I remember :D sorry

**Comment by leplatrem** [comment]
I went through the whole diff, and couldn't spot any issue.

r+ !

**Pull request merged**